source/client/factories_impl.cc

```diff
@@ -169,15 +169,15 @@ RequestSourceFactoryImpl::create(const Envoy::Upstream::ClusterManagerPtr& clust
     setRequestHeader(*header, option_header.header().key(), option_header.header().value());
   }
 
-  if (options_.requestSource() == "") {
-    return std::make_unique<StaticRequestSourceImpl>(std::move(header));
-  } else {
+  if (options_.requestSource() != "") {
```

Comment:
pre-existing, but since we edited this code:  Can we change this to be `.empty()` instead of doing empty string equality?